packages/expo-background-task/src/BackgroundTask.ts

```diff
@@ -0,0 +1,104 @@
+ * @param taskName Name of the task to register. The task needs to be defined first - see [`TaskManager.defineTask`](task-manager/#taskmanagerdefinetasktaskname-taskexecutor)
+ * for more details.
+ * @param options An object containing the background task options.
+ *
+ * @example
+ * ```ts
+ * import * as TaskManager from 'expo-task-manager';
+ *
+ * // Register the task outside of the component
+ * TaskManager.defineTask(YOUR_TASK_NAME, () => {
+ *   try {
+ *     await AsyncStorage.setItem(LAST_TASK_DATE_KEY, Date.now().toString());
+ *   } catch (error) {
+ *     console.error('Failed to save the last fetch date', error);
+ *     return BackgroundTaskResult.Failed;
+ *   }
+ *   return BackgroundTaskResult.Success;
+ * });
+ * ```
+ *
+ * You can now use the `registerTaskAsync` function to register the task:
+ *
+ * ```ts
+ * BackgroundTask.registerTaskAsync(BACKGROUND_TASK_IDENTIFIER, {});
+ * ```
+ */
+export async function registerTaskAsync(
+  taskName: string,
+  options: BackgroundTaskOptions = {}
+): Promise<void> {
+  if (!ExpoBackgroundTaskModule.registerTaskAsync) {
+    throw new UnavailabilityError('BackgroundTask', 'registerTaskAsync');
+  }
+  if (!TaskManager.isTaskDefined(taskName)) {
+    throw new Error(
+      `Task '${taskName}' is not defined. You must define a task using TaskManager.defineTask before registering.`
+    );
+  }
+  console.log('Calling ExpoBackgroundTaskModule.registerTaskAsync', { taskName, options });
+  await ExpoBackgroundTaskModule.registerTaskAsync(taskName, options);
+}
+
+// @needsAudit
+/**
+ * Unregisters a background task, so the application will no longer be executing this task.
+ * @param taskName Name of the task to unregister.
+ * @return A promise which fulfils when the task is fully unregistered.
+ */
+export async function unregisterTaskAsync(taskName: string): Promise<void> {
+  if (!ExpoBackgroundTaskModule.unregisterTaskAsync) {
+    throw new UnavailabilityError('BackgroundTask', 'unregisterTaskAsync');
+  }
+  console.log('Calling ExpoBackgroundTaskModule.unregisterTaskAsync', taskName);
+  await ExpoBackgroundTaskModule.unregisterTaskAsync(taskName);
+}
+
+// @needsAudit
+/**
+ * When in debug mode this function will trigger running the background tasks.
+ * This function will only work for apps built in debug mode.
+ * @returns A promise which fulfils when the task is triggered.
+ */
+export async function triggerTaskWorkerForTestingAsync(): Promise<void> {
+  if (__DEV__) {
+    if (!ExpoBackgroundTaskModule.triggerTaskWorkerForTestingAsync) {
+      throw new UnavailabilityError('BackgroundTask', 'triggerTaskWorkerForTestingAsync');
+    }
+    console.log('Calling triggerTaskWorkerForTestingAsync');
+    await ExpoBackgroundTaskModule.triggerTaskWorkerForTestingAsync();
+  } else {
+    return Promise.resolve();
+  }
+}
+
+// Export types
+export {
```

Comment:
```suggestion
export type {
```

Reply:
Not sure if this works - often we import like `import * as BackgroundTask from ...` and then BackgroundTask.BackgroundTaskStatus wouldn't work.